### ISSUE TITLE
Modify UI so channel selector is not avialable if only one channel.

### ISSUE THREAD

**Issue by carlkesselman** [informatics-isi-edu/microscopy#18]
Right now if there is only one image channel the channel selector still shows, even thought it does nothing.  This thould be changed so it doesn't show up unless there is more then one channel.

**Comment by howdyjessie** [comment]
@carlkesselman Does 'combo' count as a channel?

**Comment by carlkesselman** [comment]
Yes.  So we may have RGB (combo channel) which already has color assigned, so no color selector is needed, or the other channels, which are grayscale and can have color assigned to them.  Actually, I wonder if there is a good way to determine what we have, other then looking at the channel name….


Carl

----------------------------------------------------------
Dr. Carl Kesselman
Dean’s Professor, Epstein Department of Industrial and Systems Engineering
Fellow, Information Sciences Institute
Viterbi School of Engineering

Professor, Preventive Medicine
Keck School of Medicine

University of Southern California
4676 Admiralty Way, Suite 1001, Marina del Rey, CA  90292-6695
Phone:  +1 (310) 448-9338
Email:  carl@isi.edu<mailto:carl@isi.edu>
Web: http://www.isi.edu/~carl








On Nov 23, 2015, at 3:12 PM, Jessie Wong <notifications@github.com<mailto:notifications@github.com>> wrote:


@carlkesselman<https://github.com/carlkesselman> Does 'combo' count as a channel?

**Comment by meihuisu** [comment]
Both 'combo'(we can change it to RGBcombo, it is named by us anyway) and 'TL Brightfield' assume color is assigned and
the hue slider is not visible nor selectable.

mei

----- Original Message -----

**Comment by carlkesselman** [comment]
Is there a more robust way of indicating this?  In particular, I’m concerned that we will be getting pyramids generated from other microscopes, and it would be good to have our expectations well defined, and even documented so we can tell the people generating the images exactly what we expect.

Carl

----------------------------------------------------------
Dr. Carl Kesselman
Dean’s Professor, Epstein Department of Industrial and Systems Engineering
Fellow, Information Sciences Institute
Viterbi School of Engineering

Professor, Preventive Medicine
Keck School of Medicine

University of Southern California
4676 Admiralty Way, Suite 1001, Marina del Rey, CA  90292-6695
Phone:  +1 (310) 448-9338
Email:  carl@isi.edu<mailto:carl@isi.edu>
Web: http://www.isi.edu/~carl








On Nov 23, 2015, at 4:36 PM, meihuisu <notifications@github.com<mailto:notifications@github.com>> wrote:


Both 'combo'(we can change it to RGBcombo, it is named by us anyway) and 'TL Brightfield' assume color is assigned and
the hue slider is not visible nor selectable.

mei

----- Original Message -----

**Comment by karlcz** [comment]
I'm not sure what you can base it on other than one of these:

1. using channel name in the extended XML as now
2. adding explicit XML fields such as samples-per-pixel
3. learning by fetching/introspecting on a JPEG tile from the pyramid

Basically, we've only seen single-sample (grayscale) channels in
fluorescence images and 3-sample (RGB) channels in brightfield
imagery.  I don't know if/when we'll encounter anything more
unusual...

karl

**Comment by meihuisu** [comment]
I think next thing is to rework it so it is easier to extend the option without any need to edit the
source. A 2dViewer config setup or something. This is going to be true for all those greayscale/color/channelname
table or relation.

mei

----- Original Message -----

**Comment by howdyjessie** [comment]
Well anyway, here's what I've got so far: http://vm-dev-030.misd.isi.edu/~jessie/openseadragon-viewer/mview.html?url=http://vm-dev-030.misd.isi.edu/~mei/data/real3/DZC/Alexa%20Fluor%20488/ImageProperties.xml&x=0.5000000000000001&y=0.6452489905787349&z=0.6100553130613549

For now, when there's only one channel, I display the channel name in lieu of the channel selector dropdown.

**Comment by carlkesselman** [comment]
I think that is good.

There was one other issue related to contrast setting.  For the time being, we don’t want it displayed, and we need to make sure that internally the setting is at 1.  Not sure if this is the UI or app where this needs to happen.

Carl

----------------------------------------------------------
Dr. Carl Kesselman
Dean’s Professor, Epstein Department of Industrial and Systems Engineering
Fellow, Information Sciences Institute
Viterbi School of Engineering

Professor, Preventive Medicine
Keck School of Medicine

University of Southern California
4676 Admiralty Way, Suite 1001, Marina del Rey, CA  90292-6695
Phone:  +1 (310) 448-9338
Email:  carl@isi.edu<mailto:carl@isi.edu>
Web: http://www.isi.edu/~carl








On Nov 24, 2015, at 3:55 PM, Jessie Wong <notifications@github.com<mailto:notifications@github.com>> wrote:


Well anyway, here's what I've got so far: http://vm-dev-030.misd.isi.edu/~jessie/openseadragon-viewer/mview.html?url=http://vm-dev-030.misd.isi.edu/~mei/data/real3/DZC/Alexa%20Fluor%20488/ImageProperties.xml&x=0.5000000000000001&y=0.6452489905787349&z=0.6100553130613549<http://vm-dev-030.misd.isi.edu/%7Ejessie/openseadragon-viewer/mview.html?url=http://vm-dev-030.misd.isi.edu/%7Emei/data/real3/DZC/Alexa%20Fluor%20488/ImageProperties.xml&x=0.5000000000000001&y=0.6452489905787349&z=0.6100553130613549>

For now, when there's only one channel, I display the channel name in lieu of the channel selector dropdown.

**Comment by carlkesselman** [comment]
Actually, I see Mei has committed a new contrast function…so perhaps what I’m suggesting should’d be done, although defaulting to a contrast setting of 1 should still happen,

Carl

----------------------------------------------------------
Dr. Carl Kesselman
Dean’s Professor, Epstein Department of Industrial and Systems Engineering
Fellow, Information Sciences Institute
Viterbi School of Engineering

Professor, Preventive Medicine
Keck School of Medicine

University of Southern California
4676 Admiralty Way, Suite 1001, Marina del Rey, CA  90292-6695
Phone:  +1 (310) 448-9338
Email:  carl@isi.edu<mailto:carl@isi.edu>
Web: http://www.isi.edu/~carl








On Nov 25, 2015, at 7:50 AM, Carl Kesselman <carl@exchange.isi.edu<mailto:carl@exchange.isi.edu>> wrote:

I think that is good.

There was one other issue related to contrast setting.  For the time being, we don’t want it displayed, and we need to make sure that internally the setting is at 1.  Not sure if this is the UI or app where this needs to happen.

Carl

----------------------------------------------------------
Dr. Carl Kesselman
Dean’s Professor, Epstein Department of Industrial and Systems Engineering
Fellow, Information Sciences Institute
Viterbi School of Engineering

Professor, Preventive Medicine
Keck School of Medicine

University of Southern California
4676 Admiralty Way, Suite 1001, Marina del Rey, CA  90292-6695
Phone:  +1 (310) 448-9338
Email:  carl@isi.edu<mailto:carl@isi.edu>
Web: http://www.isi.edu/~carl








On Nov 24, 2015, at 3:55 PM, Jessie Wong <notifications@github.com<mailto:notifications@github.com>> wrote:


Well anyway, here's what I've got so far: http://vm-dev-030.misd.isi.edu/~jessie/openseadragon-viewer/mview.html?url=http://vm-dev-030.misd.isi.edu/~mei/data/real3/DZC/Alexa%20Fluor%20488/ImageProperties.xml&x=0.5000000000000001&y=0.6452489905787349&z=0.6100553130613549<http://vm-dev-030.misd.isi.edu/%7Ejessie/openseadragon-viewer/mview.html?url=http://vm-dev-030.misd.isi.edu/%7Emei/data/real3/DZC/Alexa%20Fluor%20488/ImageProperties.xml&x=0.5000000000000001&y=0.6452489905787349&z=0.6100553130613549>

For now, when there's only one channel, I display the channel name in lieu of the channel selector dropdown.

**Comment by howdyjessie** [comment]
Should the contrast setting be hidden only when there's 1 channel or even when there's multiple channels?

**Comment by carlkesselman** [comment]
Assuming the contrast filter is working now, it makes sense to have this for any number of channels.

Carl

----------------------------------------------------------
Dr. Carl Kesselman
Dean’s Professor, Epstein Department of Industrial and Systems Engineering
Fellow, Information Sciences Institute
Viterbi School of Engineering

Professor, Preventive Medicine
Keck School of Medicine

University of Southern California
4676 Admiralty Way, Suite 1001, Marina del Rey, CA  90292-6695
Phone:  +1 (310) 448-9338
Email:  carl@isi.edu<mailto:carl@isi.edu>
Web: http://www.isi.edu/~carl








On Nov 25, 2015, at 3:47 PM, Jessie Wong <notifications@github.com<mailto:notifications@github.com>> wrote:


Should the contrast setting be hidden only when there's 1 channel or even when there's multiple channels?
